app/src/main/java/com/yalta/viewmodel/MapViewModel.kt

```diff
@@ -46,6 +61,30 @@ class MapViewModel(
                     viewModelScope.launch(dispatcher) {
                         _locationService.sendCurrentLocation(locationResult.lastLocation)
                     }
+
+                    if (_currentRoute != null) {
+                        for (index in 0 until _currentRoute!!.points.size) {
+                            if (!_currentRoute!!.points[index].visited) {
```

Comment:
better extract all of this into some helper method

Reply:
done